src/renderer/base/renderer.hpp

```diff
@@ -131,7 +131,7 @@ namespace Microsoft::Console::Render
         std::function<void()> _pfnFrameColorChanged;
         std::function<void()> _pfnRendererEnteredErrorState;
         bool _destructing = false;
-        bool _forceUpdateViewport = true;
+        bool _forceUpdateViewport = false;

```

Comment:
this *also* scares me, since it impacts GDI and WDDM and BGFX!

Reply:
It’s more correct however since the only reason this member exists is because `EnablePainting` overwrites `_viewport` which it absolutely should not do. There’s no guarantee that the `IRenderData` has a valid viewport at that point. Because it overwrites  `_viewport` we need this Boolean to ensure `UpdateViewport` is called anyways. So in a sense this is more correct because it puts the overwrite to where the mistake is happening in the first place.